generate-osm/db-utils/map-switches.go

```diff
@@ -0,0 +1,81 @@
+package dbUtils
+
+import (
+	"strconv"
+	OSMUtil "transform-osm/osm-utils"
+)
+
+func findAndMapAnchorSwitches(
+	abschnitt *Spurplanabschnitt,
+	osm *OSMUtil.Osm,
+	anchors map[string][]*OSMUtil.Node,
+	foundAchnorCount *int,
+	optionalNewId *int,
+) {
+	for _, knoten := range abschnitt.Knoten {
+		for _, switchBegin := range knoten.WeichenAnf {
+			for _, node := range osm.Node {
+				if len(node.Tag) == 0 {
+					continue
+				}
+
+				railwayTag, _ := OSMUtil.FindTagOnNode(node, "railway")
+				refTag, _ := OSMUtil.FindTagOnNode(node, "ref")
+				name, _ := OSMUtil.FindTagOnNode(node, "name")
+
+				if railwayTag == "switch" &&
+					(refTag == switchBegin.Name.Value || name == switchBegin.Name.Value) {
+					anchors[switchBegin.Kilometrierung.Value] = append(anchors[switchBegin.Kilometrierung.Value], node)
+					newSwitchNode := createNewSwitch(optionalNewId, node, switchBegin)
+					osm.Node = append(osm.Node, &newSwitchNode)
+					*foundAchnorCount++
+				}
+			}
+		}
+
+		restSwitches := make([]*Weichenknoten, len(knoten.WeichenStamm)+len(knoten.WeichenAbzwLinks)+len(knoten.WeichenAbzwRechts))
+		copy(restSwitches, knoten.WeichenStamm)
+		copy(restSwitches[len(knoten.WeichenStamm):], knoten.WeichenAbzwLinks)
+		copy(restSwitches[len(knoten.WeichenStamm)+len(knoten.WeichenAbzwLinks):], knoten.WeichenAbzwRechts)
+		for _, switchBegin := range restSwitches {
+			for _, node := range osm.Node {
+				if len(node.Tag) == 0 {
+					continue
+				}
+
+				railwayTag, _ := OSMUtil.FindTagOnNode(node, "railway")
+				refTag, _ := OSMUtil.FindTagOnNode(node, "ref")
+				name, _ := OSMUtil.FindTagOnNode(node, "name")
+
+				if railwayTag == "switch" {
+					partnerName := switchBegin.Partner.Name
+
+					if (partnerName == refTag || partnerName == name) &&
+						anchors[switchBegin.Kilometrierung.Value] == nil {
+						anchors[switchBegin.Kilometrierung.Value] = append(anchors[switchBegin.Kilometrierung.Value], node)
```

Comment:
I just look for anchor. I only add a new switch on a WeichenAnfang, so there always only one. But we can still use this as an anchor because the kilometrage is the same on WeichenAnfang, WeichenStamm, WeichenAbzwLinks and WeichenAbzwRechts.